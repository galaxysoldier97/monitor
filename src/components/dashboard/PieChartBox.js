import React from 'react';
import PropTypes from 'prop-types';
import { Box, Chip, Grid, Paper, Typography } from '@material-ui/core';
import { grey } from '@material-ui/core/colors';
import { Cell, Pie, PieChart, ResponsiveContainer } from 'recharts';

const PieChartBox = (props) => {
  return (
    <Paper>
      <Box padding={3}>
        <Grid container spacing={2}>
          {props.name && <Grid item xs={12}>
            <Typography>{props.name}</Typography>
          </Grid>}
          <Grid item xs={12} sm={8}>
            <ResponsiveContainer height={290} width="100%">
              <PieChart>
                <Pie
                  innerRadius={70}
                  outerRadius={120}
                  data={props.data}
                  dataKey="value"
                  fill="#8884d8">
                  {
                    props.data.map((item) => <Cell key={item.name} fill={item.color} />)
                  }
                </Pie>
              </PieChart>
            </ResponsiveContainer>
          </Grid>
          <Grid item xs={12} sm={4}>
            <Grid container direction="column" alignItems="flex-end" spacing={1}>
              {props.data.map((item, index) =>
                <Grid item key={index}>
                  <Chip color="secondary" style={{backgroundColor: item.color || grey[500]}} label={`${item.value} ${item.name}`} />
                </Grid>,
              )}
            </Grid>
          </Grid>
        </Grid>
      </Box>
    </Paper>
  );
};

PieChartBox.propTypes = {
  data: PropTypes.array,
  name: PropTypes.string,
};

export default PieChartBox;
