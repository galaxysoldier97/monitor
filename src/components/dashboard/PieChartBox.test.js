import React from 'react';
import renderer from 'react-test-renderer';
import PieChartBox from './PieChartBox';
import { cyan } from '@material-ui/core/colors';

jest.mock('recharts', () => ({
  ResponsiveContainer: jest.fn,
  Cell: jest.fn,
  Pie: jest.fn,
  PieChart: jest.fn,
}));

test('PieChartBox snapshot', () => {
  const component = renderer.create(<PieChartBox data={[{name: 'test', color: cyan[100], value: 10}, {name: 'test2', color: cyan[200], value: 20}]} name={'test chart'} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
