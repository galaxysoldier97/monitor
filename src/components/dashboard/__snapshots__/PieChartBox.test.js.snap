// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChartBox snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-30"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          test chart
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-end"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiChip-root MuiChip-colorSecondary"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "backgroundColor": "#b2ebf2",
                }
              }
            >
              <span
                className="MuiChip-label"
              >
                10 test
              </span>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiChip-root MuiChip-colorSecondary"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "backgroundColor": "#80deea",
                }
              }
            >
              <span
                className="MuiChip-label"
              >
                20 test2
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
